src/serialize.js

```diff
@@ -76,16 +78,23 @@ const serializeResource = (type, resource, options, root = false) => {
   const attributeOptions = options.attributes || []
   const relationshipOptions = options.relationships || {}
   const relationshipNames = Object.keys(relationshipOptions)
+
   const {
     attributes,
     identifier,
-    relationships
+    relationships,
+    polymorphicType
   } = extractResourceInformation(resource, attributeOptions, relationshipNames)
   const {
     included,
     relationships: serializedRelationships
   } = serializeRelationships(relationships, relationshipOptions)
 
+  if (type === 'polymorphic') {
+    if (polymorphicType) type = polymorphicType
+    else throw `You did not specify a type for the resource with id ${identifier}.`
```

Comment:
`identifier` is an object, e.g. `{ name: 'lid', value: 'uuid', valid: true }`, because the identifier might be `id` or `lid`.

```suggestion
    else throw `You did not specify a type for the resource with ${identifier.name} ${identifier.value}.`
```